reactor-core/src/test/java/reactor/core/publisher/FluxTakeUntilPredicateTest.java

```diff
@@ -160,8 +176,7 @@ public void predicateThrows() {
 			throw new RuntimeException("forced failure");
 		}).subscribe(ts);
 
-		ts.assertValues(1)
-		  .assertNotComplete()
+		ts.assertNotComplete()
```

Comment:
seems to be unrelated change

Reply:
Actually the exception in the predicate here now happens before the element "1" is emitted, so "1" never gets out.

Reply:
I have doubts here. I understand the reasoning, but this is a potentially breaking change if anyone relied on this behaviour. The behaviour is not backwards compatible with regards to error handling. Assuming there is code out there that actually deals with the delivery of the item that then causes an exception when tested, we should preserve this behaviour, or address this change for 3.6 with a clear explanation this is breaking.

Reply:
The later one would be cleaner, wouldn't it? Shall I rebase? Whats the procedure on breaking changes?

Reply:
I don't think there is breaking change unless there is a bug in the given test, even though predicate result is false value is delivered since return statement happens after the a.onNext. I will have a closer look to figure out what is wrong with the test

Reply:
the onNext method should look like the following.

```java
@Override
		public void onNext(T t) {
			if (done) {
				Operators.onNextDropped(t, actual.currentContext());
				return;
			}

			boolean b;

			try {
				b = predicate.test(t);
			} catch (Throwable e) {
				// ensures value is delivered regardless the failure
				actual.onNext(t);
				onError(Operators.onOperatorError(s, e, t, actual.currentContext()));
				return;
			}

			actual.onNext(t);

			if (b) {
				s.cancel();

				onComplete();
			}
		}
```

having it like that will make the test pass. We have to preserve original behaviour